### PR TITLE
chore: security hardening

### PR DESCRIPTION
**Pull request by 633kh4ck**
Some security enhancements, among them:
- initialize objects without a prototype in keychain module
- add auditors as CODEOWNERS for CI/CD
- harden workflows' permissions
- explicitly specify CodeQL queries

### CODE REVIEW

File: features/keychain/module/memoized-keychain.js [hunk 1]
```diff
@@ -12,14 +12,14 @@ const getPublicKeyData = ({ xpub, publicKey }) => ({ xpub, publicKey })
...
-  #publicKeys = {}
+  #publicKeys = Object.create(null)
   #cloneOpts
   constructor({ storage, logger }) {
     super({ id: MODULE_ID, logger })
 
     this.#storage = storage
     this.#storage.get(CACHE_KEY).then((data) => {
-      this.#publicKeys = data ? BJSON.parse(data) : {}
+      this.#publicKeys = data ? BJSON.parse(data) : Object.create(null)
```
Comment (mvayngrib) [inline features/keychain/module/memoized-keychain.js:22]: @633kh4ck can we move this out into a separate `fix` PR for keychain?

Comment (633kh4ck) [reply features/keychain/module/memoized-keychain.js:22]: @mvayngrib, sure, will do

**Comment by mvayngrib** [comment]
@diegomura if those keychain changes look innocent to u, i'll merge this, we're coming up on a deadline 😅

**Review by diegomura — APPROVED** [review]
utACK

Changes look fine, although not sure I fully get the reasons behind it, at least for the default argument cases. We aren't passing those values anywhere, just reading predefined keys from them

**Review by mvayngrib — APPROVED** [review]

**Pull request merged**